micro-rdk/src/common/config.rs

```diff
@@ -270,17 +299,225 @@ impl TryFrom<&google::protobuf::value::Kind> for Kind {
     }
 }
 
-#[derive(Debug, Default)]
+pub struct Model {
+    familly: String,
+    model: String,
+    namespace: String,
+}
+
+impl Model {
+    pub fn new_builtin(model: String) -> Self {
+        Self {
+            familly: "builtin".to_owned(),
+            namespace: "rdk".to_owned(),
+            model,
+        }
+    }
+    pub fn get_model(&self) -> &str {
+        &self.model
+    }
+    pub fn get_familly(&self) -> &str {
+        &self.familly
+    }
+    pub fn get_namespace(&self) -> &str {
+        &self.namespace
+    }
+}
+impl Debug for Model {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_fmt(format_args!(
+            "Model({}:{}:{})",
+            self.namespace, self.familly, self.model
+        ))
+    }
+}
+// this is equivalent to the regex \w-
+fn is_valid_api_model_char(c: char) -> bool {
+    c.is_alphanumeric() || c == '_' || c == '-'
+}
+
+// for both model and api we follow the triplet constrains of RDK
+impl TryFrom<&str> for Model {
+    type Error = ConfigModelError;
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
```

Comment:
this is a bit exhaustive and we may consider using a Regex instead (like RDK) the code of doing so is unclear to me however

Reply:
haven't tested it myself, but could we not just do something like this?
```rust
let reg = Regex::new(r"<namespace>:<family>:<model>").unwrap()?
let model_components = reg.captures(value)?;
let namespace = reg.get("namespace").ok_or(...).and_then(...)?;
...
```

Reply:
Sorry I meant the cost of using a regex :P. Seems like using the regex crate blows up the binary by 700k and blow up the stack. I kinda expected it but this is way more than i anticipated :P